### PR TITLE
SUPESC-34 spryker/development not required

### PR DESCRIPTION
**Pull request by oleksandr-stoilovskyi**
- Developer(s): @oleksandr-stoilovskyi

- Ticket: https://spryker.atlassian.net/browse/SUPESC-34

- Academy PR: -

- Release Group: https://release.spryker.com/release-groups/view/3375

#### Please confirm

- [ ] No new OS components - or they have been approved by legal department

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   Process               | patch                 |            |

-----------------------------------------

##### Change log

Fixes

- Removed Development module from dependencies as require-dev packages are not allowed as require ones.

### CODE REVIEW

**Comment by oleksandr-stoilovskyi** [comment]
It looks like failed checks - it is expected behavior for this repository?

File: composer.json [hunk 1]
```diff
@@ -6,14 +6,14 @@
...
         "spryker/symfony": "^3.0.0",
         "spryker/transfer": "^3.4.0",
         "spryker/util-encoding": "^2.0.0",
         "spryker-middleware/logger": "^1.0.0"
     },
     "require-dev": {
+        "spryker/development": "^3.0.0",
         "spryker/calculation": "*",
         "spryker/cart": "*",
         "spryker/code-sniffer": "*",
```
Comment (matweew) [inline composer.json]: please remove "spryker/transfer": "*" from "require-dev", it's already requirement

Comment (oleksandr-stoilovskyi) [reply composer.json]: done

**Review by matweew — APPROVED** [review]

**Review by olehnovatskyi-spryker — APPROVED** [review]

**Review by oleksander-kiiashko — APPROVED** [review]

File: composer.json [hunk 2]
```diff
@@ -4,9 +4,9 @@
     "license": "proprietary",
     "minimum-stability": "dev",
     "require": {
-        "php": ">=7.2",
+        "php": ">=7.3",
+        "spryker/kernel": "^3.30.0",
         "spryker/console": "^4.0.0",
```
Comment (matweew) [inline composer.json]: correct ABC order
```suggestion
        "spryker/console": "^4.0.0",
        "spryker/kernel": "^3.30.0",
```

File: tests/SprykerMiddlewareTest/Zed/Process/Business/ProcessFacadeTest.php [hunk 3]
```diff
@@ -86,7 +88,7 @@ protected function getProcessSettingsTransfer(): ProcessSettingsTransfer
         $transfer->setInputPath(static::PATH_INPUT);
         $transfer->setOutputPath(static::PATH_OUTPUT);
         $transfer->setIteratorConfig(new IteratorConfigTransfer());
-        $transfer->setLoggerConfig(new LoggerConfigTransfer());
+        $transfer->setLoggerConfig((new LoggerConfigTransfer())->setVerboseLevel(100));
```
Comment (matweew) [inline tests/SprykerMiddlewareTest/Zed/Process/Business/ProcessFacadeTest.php]: why is it 100?
Logger::DEBUG ?

File: tests/SprykerMiddlewareTest/Zed/Process/Business/Mapper/MapperTest.php [hunk 4]
```diff
@@ -8,6 +8,7 @@
 namespace SprykerMiddlewareTest\Zed\Process\Business\Mapper;
 
 use Codeception\Test\Unit;
+use Everon\Component\Factory\Tests\Unit\Doubles\LoggerStub;
```
Comment (matweew) [inline tests/SprykerMiddlewareTest/Zed/Process/Business/Mapper/MapperTest.php]: not used?

File: tests/SprykerMiddlewareTest/Zed/Process/Business/ProcessFacadeTest.php [hunk 5]
```diff
@@ -33,6 +33,8 @@
  * @group Business
  * @group ProcessFacade
  * @group ProcessFacadeTest
+ *
+ * @property \SprykerMiddlewareTest\Zed\Process\ProcessZedTester $tester
```
Comment (matweew) [inline tests/SprykerMiddlewareTest/Zed/Process/Business/ProcessFacadeTest.php]:
```
    /**
     * @var \SprykerMiddlewareTest\Zed\Process\ProcessZedTester
     */
    protected $tester;
```

File: tests/SprykerMiddlewareTest/Zed/Process/Business/Mapper/MapperTest.php [hunk 6]
```diff
@@ -35,28 +36,24 @@
  * @group Business
  * @group Mapper
  * @group MapperTest
+ *
+ * @property \SprykerMiddlewareTest\Zed\Process\ProcessZedTester $tester
```
Comment (matweew) [inline tests/SprykerMiddlewareTest/Zed/Process/Business/Mapper/MapperTest.php]:
```

	/**
     * @var \SprykerMiddlewareTest\Zed\Process\ProcessZedTester
     */
    protected $tester;

**Review by matweew — CHANGES_REQUESTED** [review]

File: composer.json (hunk 2 above)
Comment (oleksandr-stoilovskyi) [reply composer.json]: fixed

File: tests/SprykerMiddlewareTest/Zed/Process/Business/ProcessFacadeTest.php (hunk 3 above)
Comment (oleksandr-stoilovskyi) [reply tests/SprykerMiddlewareTest/Zed/Process/Business/ProcessFacadeTest.php]: fixed

File: tests/SprykerMiddlewareTest/Zed/Process/Business/Mapper/MapperTest.php (hunk 4 above)
Comment (oleksandr-stoilovskyi) [reply tests/SprykerMiddlewareTest/Zed/Process/Business/Mapper/MapperTest.php]: deleted

File: tests/SprykerMiddlewareTest/Zed/Process/Business/ProcessFacadeTest.php (hunk 5 above)
Comment (oleksandr-stoilovskyi) [reply tests/SprykerMiddlewareTest/Zed/Process/Business/ProcessFacadeTest.php]: fixed

File: tests/SprykerMiddlewareTest/Zed/Process/Business/Mapper/MapperTest.php (hunk 6 above)
Comment (oleksandr-stoilovskyi) [reply tests/SprykerMiddlewareTest/Zed/Process/Business/Mapper/MapperTest.php]: fixed

**Review by matweew — APPROVED** [review]

**Comment by dereuromark** [comment]
Please adjust changelogs:
Require dev changes are not relevant for customers
Instead, some other functional changes are missing, like PHP8 comp. and Kernel update for setter/getter usage on the stacks

**Comment by dereuromark** [comment]
@matweew Whats the status on this? This is an important release as require-dev packages sneaking into prod is a bad and insecure thing.

**Comment by matweew** [comment]
@dereuromark I'm not the part of SUPESC stream, so cannot answer this question.
It was approved from my side a long time ago.

The better question is what is the status of https://spryker.atlassian.net/browse/SUPESC-34

(unfortunately I cannot tag somebody from SCOS stream here)

**Comment by floriankraemer** [comment]
@matweew you can't tag me?

**Comment by matweew** [comment]
forgot about you @floriankraemer, I can :)

**Review by dereuromark — APPROVED** [review]
This must be released ASAP
The development module dependency is also a security concern.

**Pull request merged**